test/integration/api/domain_admin_contacts_test.rb

```diff
@@ -0,0 +1,133 @@
+require 'test_helper'
+
+class APIDomainAdminContactsTest < ApplicationIntegrationTest
+  def test_replace_all_admin_contacts_of_the_current_registrar
+    patch '/repp/v1/domains/admin_contacts', params: { current_contact_id: 'william-001',
+                                                 new_contact_id: 'john-001' },
+          headers: { 'HTTP_AUTHORIZATION' => http_auth_key }
+
+    assert_nil domains(:shop).admin_contacts.find_by(code: 'william-001')
```

Comment:
And here we are facing the problem.
At the start of the method we do not have contact with current_contact_id `william-001` as an admin contact of domain `:shop`.
![image](https://user-images.githubusercontent.com/7688888/106259849-93fa9000-6241-11eb-96e2-f0f9abc0f5cb.png)
So the whole procedure of replacing contacts looses sense since we are replacing wrong contacts.

So my proposal is the following:
1) At the start of any test method let's make sure the start data and conditions are correct (by using breakpoints like `binding.pry` and checking data like `domains(:shop).admin_contacts` and such).
2) We do need as well the test to checking if `Contact objects are replaced only if ident data of old and new object match (ident, ident type and ident country code)`, I cannot find any ATM. Let's add this.